### PR TITLE
Adds event middleware to the docs

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

File: plugins/event-middleware.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+use League\Tactician\CommandEvents\EventMiddleware;
+
+$emitter = new Emitter;
+$eventMiddleware = new EventMiddleware($emitter);
+
+// other possible solution
+// $eventMiddleware->setEmitter($emitter);
+~~~
+
+You can also catch an error and prevent it from causing the application fail:
```
Comment (rosstuck) [inline plugins/event-middleware.md]: Little typo: "prevent it from causing the application **to** fail:"

File: plugins/event-middleware.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+title: Event Middleware
+---
+
+# Event Middleware
+
+[![Author](http://img.shields.io/badge/author-@sagikazarmark-blue.svg?style=flat-square)](https://twitter.com/sagikazarmark)
+[![Source](http://img.shields.io/badge/source-league/tactician--command--events-blue.svg?style=flat-square)](https://github.com/thephpleague/tactician-command-events)
+[![Packagist](http://img.shields.io/packagist/v/league/tactician-command-events.svg?style=flat-square)](https://packagist.org/packages/league/tactician-command-events)
+
+This plugin lets you to listen to some events emitted during command execution:
```
Comment (rosstuck) [inline plugins/event-middleware.md]: Little typo: "lets you ~~to~~ listen"

**Comment by rosstuck** [comment]
Looking good! One last request: can you change the code samples to be inline with PSR-2? Specifically adding () brackets after "new ClassName" and wrapping the functions onto new lines? I don't always like it either but I reckon it's better to be consistent in the docs.

**Comment by sagikazarmark** [comment]
Is it in the PSR? Having () after class instantiation even if there is no argument list? I can't find any clue for that. If it is explicitly in the standard then so be it. Honestly, It rather bothers me.

**Comment by sagikazarmark** [comment]
> wrapping the functions onto new lines?

I am not sure what you mean.

**Comment by rosstuck** [comment]
Paranthese are definitely the expected style and used everywhere else, so please add them. :)

PSR-2 does definitely require a closure being passed like this

```php
$eventMiddleware->addListener(
    'command.failed',
    function (CommandFailed $event) {
        $event->catchException(); // without calling this the exception will be thrown
    }
);
```

(Also, note the space after `function` and before the opening bracket)

**Comment by sagikazarmark** [comment]
Done. Yeah, the space is a good point, didn't notice.

**Pull request merged**